internal/controller/admission/falconadmission_controller_test.go

```diff
@@ -10,82 +10,137 @@ import (
+				Expect(k8sClient.Delete(ctx, &item)).To(Succeed())
+			}
+
+			Eventually(func() int {
+				deployList := &appsv1.DeploymentList{}
+				_ = k8sClient.List(ctx, deployList, client.InNamespace(namespaceName))
+				return len(deployList.Items)
+			}, 6*time.Second, 2*time.Second).Should(Equal(0))
+
+			// Delete cluster level resources
+			clusterRoleBinding := &rbacv1.ClusterRoleBinding{}
+			Expect(k8sClient.Get(ctx, types.NamespacedName{Name: admissionClusterRoleBindingName}, clusterRoleBinding)).To(Succeed())
+			Expect(k8sClient.Delete(ctx, clusterRoleBinding)).To(Succeed())
+
+			if admissionControlEnabled {
+				validatingWebhookConfig := &arv1.ValidatingWebhookConfiguration{}
+				Expect(k8sClient.Get(ctx, types.NamespacedName{Name: ValidatingWebhookName}, validatingWebhookConfig)).To(Succeed())
+				Expect(k8sClient.Delete(ctx, validatingWebhookConfig)).To(Succeed())
+			}
+
+			// Delete FalconAdmission custom resource
+			falconAdmissionCR := &falconv1alpha1.FalconAdmission{}
+			Expect(k8sClient.Get(ctx, typeNamespaceName, falconAdmissionCR)).To(Succeed())
+			Expect(k8sClient.Delete(ctx, falconAdmissionCR)).To(Succeed())
+
 			// TODO(user): Attention if you improve this code by adding other context test you MUST
 			// be aware of the current delete namespace limitations. More info: https://book.kubebuilder.io/reference/envtest.html#testing-considerations
 			By("Deleting the Namespace to perform the tests")
 			_ = k8sClient.Delete(ctx, namespace)
 		})
 
+		// Validating a generic deployment of FalconAdmission
 		It("should successfully reconcile a custom resource for FalconAdmission", func() {
-			By("Creating the custom resource for the Kind FalconAdmission")
-			falconAdmission := &falconv1alpha1.FalconAdmission{}
-			err := k8sClient.Get(ctx, typeNamespaceName, falconAdmission)
-			if err != nil && errors.IsNotFound(err) {
-				// Let's mock our custom resource at the same way that we would
-				// apply on the cluster the manifest under config/samples
-				falconAdmission := &falconv1alpha1.FalconAdmission{
-					ObjectMeta: metav1.ObjectMeta{
-						Name:      AdmissionControllerName,
-						Namespace: AdmissionControllerNamespace,
-					},
-					Spec: falconv1alpha1.FalconAdmissionSpec{
-						Falcon: falconv1alpha1.FalconSensor{
-							CID: &falconCID,
-						},
-						InstallNamespace: "falcon-kac",
-						Image:            admissionImage,
-						Registry: falconv1alpha1.RegistrySpec{
-							Type: "crowdstrike",
-						},
-						AdmissionConfig: falconv1alpha1.FalconAdmissionConfigSpec{
-							DepUpdateStrategy: falconv1alpha1.FalconAdmissionUpdateStrategy{
-								RollingUpdate: appsv1.RollingUpdateDeployment{
-									MaxUnavailable: &intstr.IntOrString{IntVal: 1},
-									MaxSurge:       &intstr.IntOrString{IntVal: 1},
-								},
-							},
-						},
-					},
-				}
+			admissionControlEnabled = true
+			falconAdmission.Spec.AdmissionConfig.AdmissionControlEnabled = &admissionControlEnabled
```

Comment:
major: Can we leave out line 139 to make sure the default behavior is to enable admission control. Even if `falconAdmission.Spec.AdmissionConfig.AdmissionControlEnabled` is nil, admission control should be enabled.

Reply:
Updated and validates defaults.